Userland/Libraries/LibJS/Runtime/Temporal/Calendar.cpp

```diff
@@ -67,6 +68,216 @@ ReadonlySpan<StringView> available_calendars()
+    }
+        JS_ENUMERATE_CALENDAR_METHODS
+#undef __JS_ENUMERATE
+    }
+
+    // 10. Return unused.
+    return {};
+}
+
+// 12.2.5 CalendarMethodsRecordHasLookedUp ( calendarRec, methodName ), https://tc39.es/proposal-temporal/#sec-temporal-calendarmethodsrecordhaslookedup
+bool calendar_methods_record_has_looked_up(CalendarMethodsRecord const& calendar_record, CalendarMethod method_name)
+{
+    // 1. If methodName is DATE-ADD, then
+    //     a. Let method be calendarRec.[[DateAdd]].
+    // 2. Else if methodName is DATE-FROM-FIELDS, then
+    //     a. Let method be calendarRec.[[DateFromFields]].
+    // 3. Else if methodName is DATE-UNTIL, then
+    //     a. Let method be calendarRec.[[DateUntil]].
+    // 4. Else if methodName is DAY, then
+    //     a. Let method be calendarRec.[[Day]].
+    // 5. Else if methodName is FIELDS, then
+    //     a. Let method be calendarRec.[[Fields]].
+    // 6. Else if methodName is MERGE-FIELDS, then
+    //     a. Let method be calendarRec.[[MergeFields]].
+    // 7. Else if methodName is MONTH-DAY-FROM-FIELDS, then
+    //     a. Let method be calendarRec.[[MonthDayFromFields]].
+    // 8. Else if methodName is YEAR-MONTH-FROM-FIELDS, then
+    //     a. Let method be calendarRec.[[YearMonthFromFields]].
+    // 9. If method is undefined, return false.
+    // 10. Return true.
+    switch (method_name) {
+#define __JS_ENUMERATE(PascalName, camelName, snake_name) \
+    case CalendarMethod::PascalName: {                    \
+        return calendar_record.snake_name != nullptr;     \
+    }
+        JS_ENUMERATE_CALENDAR_METHODS
+#undef __JS_ENUMERATE
+    }
+    VERIFY_NOT_REACHED();
+}
+
+// 12.2.6 CalendarMethodsRecordIsBuiltin ( calendarRec ), https://tc39.es/proposal-temporal/#sec-temporal-calendarmethodsrecordisbuiltin
+bool calendar_methods_record_is_builtin(CalendarMethodsRecord const& calendar_record)
+{
+    // 1. If calendarRec.[[Receiver]] is a String, return true.
+    if (calendar_record.receiver.has<String>())
+        return true;
+
+    // 2. Return false.
+    return false;
+}
+
+// 12.2.7 CalendarMethodsRecordCall ( calendarRec, methodName, arguments ), https://tc39.es/proposal-temporal/#sec-temporal-calendarmethodsrecordcall
+ThrowCompletionOr<Value> calendar_methods_record_call(VM& vm, CalendarMethodsRecord const& calendar_record, CalendarMethod method_name, Span<Value> arguments)
+{
+    // 1. Assert: CalendarMethodsRecordHasLookedUp(calendarRec, methodName) is true.
+    VERIFY(calendar_methods_record_has_looked_up(calendar_record, method_name));
+
+    // 2. Let receiver be calendarRec.[[Receiver]].
+    // 3. If CalendarMethodsRecordIsBuiltin(calendarRec) is true, then
+    //     a. Set receiver to ! CreateTemporalTimeZone(calendarRec.[[Receiver]]).
+    GCPtr<Object> receiver;
+    if (calendar_record.receiver.has<String>())
```

Comment:
Call the AO to make it clear we didn't deviate?
```c++
if (calendar_methods_record_is_builtin(calendar_record))
```

Reply:
Sure thing, I was considering which one to do at the time :^)